lib/Byng/Pimcore/Sitemap/Generator/SitemapGenerator.php

```diff
@@ -65,18 +66,51 @@ public function __construct()
      */
     public function generateXml()
     {
-        // Get all the root elements with parentId '1'
-        $rootDocuments = $this->documentGateway->getChildren(1);
-
-        foreach ($rootDocuments as $rootDocument) {
-            $this->addUrlChild($rootDocument);
-            $this->listAllChildren($rootDocument);
+        // Retrieve site trees
+        $siteRoots = new Site\Listing();
+        $siteRoots = $siteRoots->load();
+
+        // Build siteRoots table: [ ID => Domain ]
+        /* @var Site $siteRoot */
+        foreach ($siteRoots as $siteRoot){
+            $this->sitesRoots[$siteRoot->getRootId()] = $siteRoot->getMainDomain();
         }
-        $this->xml->asXML(PIMCORE_DOCUMENT_ROOT . "/sitemap.xml");
 
-        if (Config::getSystemConfig()->get("general")->get("environment") === "production") {
-            $this->notifySearchEngines();
+        // Also append the default tree
+        $this->sitesRoots[1] = Config::getSystemConfig()->get("general")->get("domain");
+
+        $notifySearchEngines = Config::getSystemConfig()->get("general")->get("environment") === "production";
+        foreach($this->sitesRoots as $siteRootID => $siteRootDomain){
+            $this->generateSiteXml($siteRootID, $siteRootDomain);
+
+            if ($notifySearchEngines)
+                $this->notifySearchEngines('https://' . $siteRootDomain);
         }
+
+    }
+
+    /**
+     * Generate a sitemap xml file for a defined site, with a specific hostUrl
+     *
+     * @param $rootId
+     * @param $hostUrl
+     * @return string
+     */
+    private function generateSiteXml($rootId, $hostUrl)
+    {
+        $this->xml = new SimpleXMLElement(
+            '<?xml version="1.0" encoding="UTF-8"?>'
+            . '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"></urlset>'
+        );
+
+        // Set current hostUrl
+        $this->hostUrl = 'https://' . $hostUrl;
```

Comment:
Alright, I tried the configuration file way here: https://github.com/byng-systems/pimcore-sitemap-plugin/pull/6/commits/98fb54b0bce524a8a9bfbfe467e6c6611e23a87b

At the installation, we list all the sites and try to reach the domains using https to determine the right protocol to use. This configuration is stored in an editable XML with other useful parameters (domain, id, path). This XML is then used to generate the sitemaps (instead of fetching the list of available sites every time).

The major drawback of this approach is every time new sites are added or site domains are changed, the XML has to be changed manually (or the admin should uninstall/reinstall the plugin). 
But it is definitely more flexible than presuming https will be used every time. Furthermore, it allows admins to choose `http` as favourite protocol even if their domain handles `https`.